.jenkins/compilation.stages

```diff
@@ -10,6 +10,9 @@ stage('Compile Downstream Build') {
         def gitURL = env.ghprbAuthorRepoGitUrl ?: env.GIT_URL
         def project = util.getProjectGroupName(util.getProject(gitURL))[1]
         projectCollection.removeAll(['droolsjbpm-tools', 'kie-docs'])
-        compile.build(projectCollection, project, "${SETTINGS_XML_ID}", "${PROPERTIES_FILE_ID}")
+        def propContent = readFile PROPERTIES_FILE_PATH
```

Comment:
I would consider moving this logic to the Jenkins shared library either into the `getGoals` function or even better as another function set like `propertiesFromPath(...)`, `propertiesFromFileId(...)`, etc., and use that to supply the required parameters from here.

Reply:
In case we do it we should pass absolute path to it, something like `"${env.WORKSPACE}/.jenkins/compileDownstreamBuild.properties`. `util` script would be the proper place for this.
There is one problem here. The `.jenkins` (or `.ci`) folder can be `"${env.WORKSPACE}/.jenkins/compileDownstreamBuild.properties` or `"${env.WORKSPACE}/droolsjbpm-build-bootstrap/.jenkins/compileDownstreamBuild.properties`

- https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/aa2723e865b8bb0b315827f5e9e5083c39ab9aa3/Jenkinsfile#L49
- https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/aa2723e865b8bb0b315827f5e9e5083c39ab9aa3/Jenkinsfile#L59

In this case we should wrap stage files content in a method in order to pass folder as an argument
```
def execute(def ciFolderPath)
stage('New Downstream Build') {
...
...
...
final PROPERTIES_FILE_PATH = "${ciFolderPath}/compileDownstreamBuild.properties"
...
...
...
}
```
and change load calling in case it is in `WORKSPACE` folder https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/aa2723e865b8bb0b315827f5e9e5083c39ab9aa3/Jenkinsfile#L49
```
def stage = load("$WORKSPACE/${file}")
stage.execute(''$WORKSPACE/.jenkins")

```
and in case it is `droolsjbpm-build-bootstrap` https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/aa2723e865b8bb0b315827f5e9e5083c39ab9aa3/Jenkinsfile#L59
```
def stage = load("${file}")
stage.execute(''$WORKSPACE/droolsjbpm-build-bootstrap/.jenkins")
```